modules/tensor_mechanics/test/tests/strain_energy_density/tests

```diff
@@ -36,7 +36,7 @@
    input = 'tot_model.i'
    cli_args = 'Materials/strain_energy_density/incremental=true'
    design = 'StrainEnergyDensity.md'
-   expect_err = 'One or more Material Properties were not supplied on block'
+   expect_err = 'Material property 'strain_increment', requested by'
```

Comment:
I don't understand how the changes in this PR could lead to this error message being different.  All I see are changes that affect how many missing properties are reported.

Reply:
The error check that reports "One or more Material Properties were not supplied on block" happens after the error check that reports "Material property 'strain.....". The problem is that the error check for the latter was not being done properly because the storage of what materials were defined also included the materials that were requested.

Reply:
Okay - I see it now.  The way the diff was shown - with context/scope hidden - it didn't look like you changed what was called in any effectual way.  But now I see you moved the checks inside of an if statement.  The unexpanded diff just made it look like you moved the calls up in the function (within the same scope).  [facepalm]